decoder/DecoderApi_unittest.cpp

```diff
@@ -273,7 +273,10 @@ TEST_P(DecodeApiTest, Format_Change)
     SharedPtr<IVideoDecoder> decoder;
     TestDecodeFrames frames = *GetParam();
     decoder.reset(createVideoDecoder(frames.getMime()), releaseVideoDecoder);
-    ASSERT_TRUE(bool(decoder));
+    if (!bool(decoder)) {
+        delete allocator;
+        ASSERT_TRUE(bool(decoder));
+    }
```

Comment:
It would be better to just declare and create the `allocator` after asserting the decoder... that way we can just keep it simple with the original `ASSERT_TRUE(bool(decoder))`.

Also, where does the `allocator` get deleted on successful test?  Does the `decoder` take ownership and delete it?

Reply:
@uartie there is a line "m_externalAllocator.reset(allocator,unrefAllocator);", it should take ownership the variable, so the system will delete the memory after the tests complete.